python_modules/dagster-graphql/dagster_graphql/schema/pipelines/pipeline.py

```diff
@@ -101,6 +102,34 @@ class Meta:
         name = "MaterializationCountByPartition"
 
 
+class GrapheneAssetObservation(graphene.ObjectType):
```

Comment:
Again, what's the difference between ObservationOrMaterialization, ObservationEvent, and AssetObservation?

Reply:
ObservationOrMaterialization is the actual AssetObservation object, so the metadata contained within the asset observation. ObservationEvent contains the event metadata (e.g. which step yielded the observation, the step stats) as well as a reference to the AssetObservation object.

I followed the pattern that we used for AssetMaterialization for AssetObservation, maybe to surface partition and run ID? Not really sure why we had this, but I can modify ObservationEvent to contain the run id and partition so we can remove AssetObservation.